### PR TITLE
Jsonpart

### PR DESCRIPTION
**Pull request by ramkivenkatachalam**
```
Secor offset check/fix tool
```

### CODE REVIEW

File: src/main/java/com/pinterest/secor/main/OffsetCheckMain.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+ *         -cp "secor-0.1-SNAPSHOT.jar:lib/*" com.pinterest.secor.main.OffsetCheckMain --fix
+ *
+ * @author Ramki Venkatachalam (ramki@pinterest.com)
+ */
+public class OffsetCheckMain {
+    private static final Logger LOG = LoggerFactory.getLogger(OffsetCheckMain.class);
+
+    public static void main(String[] args) {
+        boolean fix = false;
+        if (args.length == 1 && args[0].contentEquals("--fix")) {
```
Comment (pgarbacki) [inline src/main/java/com/pinterest/secor/main/OffsetCheckMain.java]: Could you do this with Options (see ZookeeperClientMain for an example)?

File: src/main/java/com/pinterest/secor/common/KafkaClient.java [hunk 2]
```diff
@@ -111,7 +111,29 @@ private long findLastOffset(TopicPartition topicPartition, SimpleConsumer consum
         return offsets[0] - 1;
     }
 
-    private Message getMessage(TopicPartition topicPartition, long offset,
+  public long earliestFirstOffset(TopicPartition topicPartition, SimpleConsumer consumer) {
```
Comment (pgarbacki) [inline src/main/java/com/pinterest/secor/common/KafkaClient.java]: Fix indentation.

File: src/main/java/com/pinterest/secor/main/OffsetCheckMain.java [hunk 3]
```diff
@@ -0,0 +1,58 @@
+/**
```
Comment (pgarbacki) [inline src/main/java/com/pinterest/secor/main/OffsetCheckMain.java]: I'm wondering if we couldn't make this part of the ZookeeperClient? Conceptually it fits as it fixes data in ZK. What do you think?

File: src/main/java/com/pinterest/secor/tools/OffsetCheck.java [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+
+/**
+ * OffsetCheck compares secor committed offsets to kafka offsets and lets you fix secor
+ * offsets to earliest kafka offset.
+ *
+ * @author Ramki Venkatachalam (ramki@pinterest.com)
+ */
+public class OffsetCheck {
+
+  private static final Logger LOG = LoggerFactory.getLogger(OffsetCheck.class);
```
Comment (pgarbacki) [inline src/main/java/com/pinterest/secor/tools/OffsetCheck.java]: I think the default indentation for the project is 4 spaces.

File: src/main/java/com/pinterest/secor/tools/OffsetCheck.java [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+
+    List<String> topicsList = new ArrayList<String>();
+    TopicMetadataRequest req = new TopicMetadataRequest(topicsList);
+    kafka.javaapi.TopicMetadataResponse resp = consumer.send(req);
+    consumer.close();
+    List<kafka.javaapi.TopicMetadata> topicMetadataList = resp.topicsMetadata();
+    for (kafka.javaapi.TopicMetadata topicMetadata : topicMetadataList) {
+        final String topic = topicMetadata.topic();
+        if (!topic.matches(mConfig.getKafkaTopicFilter())) {
+        LOG.info("skipping topic " + topic);
```
Comment (pgarbacki) [inline src/main/java/com/pinterest/secor/tools/OffsetCheck.java]: Indentation seems totally messed up.

File: src/main/java/com/pinterest/secor/tools/OffsetCheck.java [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+          // no messages in kafka
+          continue;
+        }
+
+        // get committed offset from zookeeper
+        long committedOffset = -1;
+        try {
+          committedOffset = mZookeeperConnector.getCommittedOffsetCount(topicPartition);
+        } catch (Exception e) {
+          //LOG.warn("No committed offset found for {}", topicPartition);
```
Comment (pgarbacki) [inline src/main/java/com/pinterest/secor/tools/OffsetCheck.java]: Uncomment or remove.

**Comment by pgarbacki** [comment]
@ramkivenkatachalam could you address the comments?

**Comment by derekargueta** [comment]
@pgarbacki are these changes still desired? If so I can fork @ramkivenkatachalam's branch and put in another PR with the changes in the comments.

**Comment by pgarbacki** [comment]
@derekargueta yes, I think this tool would be useful. Thanks for looking into this.

**Comment by HenryCaiHaiying** [comment]
This PR would need to sync with the latest code.

**Pull request closed without merging**